src/zappy_gui_src/Core/SFML/Map/Map.cpp

```diff
@@ -48,21 +51,65 @@ void Map::_updateMoveMap(sf::Vector2f &moveMap)
 void Map::display()
 {
     sf::Vector2i mouse = sf::Mouse::getPosition(*_window.get());
-    std::size_t index = 0;
+    std::size_t index = -1;
     sf::Vector2f moveMap = {0, 0};
+    sf::FloatRect area;
+    std::vector<int> times(6);
+    sf::Clock clock;
+    std::size_t tmp = 0;
 
+    times[0] = clock.getElapsedTime().asMicroseconds();
+    clock.restart();
     _updateMoveMap(moveMap);
+    times[1] = clock.getElapsedTime().asMicroseconds();
+    clock.restart();
     for (auto &it : _tile) {
-        if (it->isOnTile(mouse)) {
-            _tileHover = index;
+        if (moveMap.x || moveMap.y)
+            it->setOrigin(it->getOrigin() + moveMap);
+    }
+    times[2] = clock.getElapsedTime().asMicroseconds();
+    times[3] = 0;
+    times[4] = 0;
+    times[5] = 0;
+    for (std::size_t i = 0; i < _tile.size(); i++) {
+        times[3] += clock.getElapsedTime().asMicroseconds();
+        clock.restart();
+        area = _tile[i]->getIsoPosition();
+        if (area.left > _window->getSize().x && area.top > _window->getSize().y)
+            continue;
+        if (area.left < 0 && area.top > _window->getSize().y)
+            continue;
+        if (area.left + area.width < 0)
+            continue;
+        if (area.left > _window->getSize().x && tmp) {
+            i = tmp + _mapSize.x - 2;
+            tmp = 0;
+            continue;
+        }
+        if (area.top + area.height < 0)
+            continue;
+        if (area.top > _window->getSize().y && tmp) {
+            i = tmp + _mapSize.x - 2;
+            tmp = 0;
+            continue;
```

Comment:
I'm not sure but what happen if not ?

Reply:
laaaaaaaaaaaaaaag

Reply:
all tile with a position in one of these condition are not allowed to be display, so they are skipped to increase the performance